CONTRIBUTING.md

```diff
@@ -1,111 +1,190 @@
-   tooling limitations, a Meta employee is required to merge the pull request.
-
-## Presto’s SQL Functions
-
-Here are specific guidelines for contributing Presto SQL functions.
+7. Iterate on this process until your changes are reviewed and accepted by a 
+   maintainer. At this point, a Meta employee will be required to merge your PR,
+   due to tooling limitations.
+
+## Coding Best Practices
+
+When submitting code contributions to Velox, make sure to adhere to the
+following best practices:
+
+1. **Coding Style**: Review and strictly follow our coding style document, 
+   available in [`CODING_STYLE.md`](CODING_STYLE.md).
+   * Velox favors consistency over personal preference. If there are 
+   technical reasons why a specific guideline should not be followed, 
+   please start a separate discussion with the community to update the
+   coding style document first.
+   * If you are simply updating code that did not comply with the coding 
+   style, please do so in a standalone PR isolated from other logic changes. 
+
+2. **Small Incremental Changes**: If the change large, work with the maintainers
+   on a plan to break and submit it as smaller (yet atomic) parts.
+   * [Research indicates](https://smartbear.com/learn/code-review/best-practices-for-peer-code-review/) 
+   that engineers can only effectively review up to 
+   400 LOC at a time. The human brain can only process so much information 
+   at a time; beyond that threshold the ability to find bugs and other flaws
+   decreases.
+   * As larger PRs usually take longer to review and iterate, they 
+   tend to slow down the software development process. As much as possible, 
+   split your work into smaller changes. 
+
+3. **Unit tests**: With rare exceptions, every PR should contain unit tests
+   covering the logic added/modified.
+   * Unit tests protect our codebase from regressions, promote less coupled
+   APIs, and provide an executable form of documentation that’s useful for
+   new engineers reasoning about the codebase.
+   * Good unit tests are fast, isolated, repeatable, and exercise all APIs 
+   including their edge cases.
+   * The lack of existing tests is not a good reason not to add tests to
+   your PR. If a component or API does not have a corresponding 
+   unit test suite, please consider improving the codebase by first adding a
+   new unit test suite to ensure the existing behavior is correct.
+
+4. **Code Comments**: Appropriately add comments to your code and document APIs.
+   * As a library, Velox code should be optimized for the reader, not the
+   writer.
+   * Comments should capture information that was on the writer’s mind, but
+   couldn’t be represented as code. 
+   * The overall goal is to make the code more obvious and remove obscurity.
+   * As a guideline, every file, class, member variable, and member function
+   that is not a getter/setter should be documented.
+   * For functions with large bodies, a good practice is to group blocks of
```

Comment:
nit: perhaps, mention that such functions should be rare and suggest to break up the function first.